### DISCUSSION TITLE
Code Actions

### DISCUSSION THREAD

**Discussion by andrew-t-james-wc** [LazyVim/LazyVim#3597] [Q&A]
When calling `vim.lsp.buf.code_action()` I get all the code actions available for the current buffer, I am curious how I can also get other actions such as gitsigns or `vtsls` actions piped in to the list? Its probably something simple I am over looking but couldn't seem to figure that out.
 
<img width="795" alt="image" src="https://github.com/LazyVim/LazyVim/assets/153027330/cf5c3ec1-12e7-4702-b631-97b7dee578d5">

### Compared to `<leader>cA`

<img width="814" alt="image" src="https://github.com/LazyVim/LazyVim/assets/153027330/a9bf4f7d-c442-443a-af5e-65929f9bd776">

**Comment by gtapplec** [comment]
@andrew-t-james-core I was wondering the same thing. It appears it was added in the following commit https://github.com/LazyVim/LazyVim/commit/9f5a0f1bd847663deea9cb48a178b85d7bdb3ec6#diff-be2205f150691c19f8618f12cea372c5c1fcba82458a67cd02d1d1a84a626ca4R34

```lua
vim.lsp.buf.code_action({context = {only = {"source"}})
```

Looking at the LSP spec a little more (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/) there appear to be multiple categories. 

From what I could gather the default seems to use "refactor" & "quickfix" from a glance. Hopefully this helps you 👋
